doc/proposal/monitoring.md

```diff
@@ -0,0 +1,144 @@
+# Design Proposal: Monitoring the Istio Service Mesh
+
+Status: **DRAFT**
+
+- [Introduction](#Introduction)
+- [Proposed Design](#Proposed_Design)
+  - [Design Summary](#Design_Summary)
+  - [Telemetry Generation And Reporting](#Telemetry_Generation_And_Reporting)
+    - [Schema Publishing](#Schema_Publishing)
+    - [Example Library Pseudocode](#Example_Library_Pseudocode)
+  - [Mixer-Specific Telemetry](#Mixer-Specific_Telemetry)
+
+## Introduction
+
+The [Istio Service Mesh](http://github.com/istio/istio) provides automated 
+telemetry data collection for integrated services. As requests flow through the
+service mesh, request metadata is generated by the [Proxy](http://github.com/istio/proxy) 
+and sent to the [Mixer](http://github.com/istio/mixer). The Mixer repackages the
+received metadata, generating access logs and request metrics, and transmits the
+telemetry data to a configurable set of backend systems for monitoring and 
+alerting.
+
+Istio development has, so far, focused on providing the subsystems for the 
+reporting of telemetry data and for tailoring the collection to match the intent
+of the service operators. Little attention has been devoted to monitoring the 
+behavior of the individual Istio components (Proxy, [Manager](http://github.com/istio/manager),
+Mixer).
+
+This document proposes a high-level design for the monitoring of the Istio 
+Service Mesh itself.
+
+## Proposed Design
+
+The proposed design adopts two high-level goals:
+1. **Limit special handling for Istio components**
+   - Telemetry data for the Istio Proxy and Istio Manager components _should_ be
+     reported via the mixer through the same mechanisms as telemetry data for 
+     the non-Istio services within the service mesh.
+   - Telemetry data for the Istio Mixer itself _should_ be sent through the same 
+     processing code as data received via RPCs.
+1. **Prevent mixer-generated traffic from generating additional reports**
+   - In order to prevent a never-ending loop of reporting of metrics, traffic
+     generated by mixer to downstream services (e.g. logging/metrics backends)
+     _must_ be exempted from triggering the push of new telemetry data to 
+     backend systems.
+
+### Design Summary
+
+Each Istio component will be responsible for the following:
+1. Generating its own set of telemetry data (logs/metrics)
+1. Reporting that data to the Istio Mixer, on a regular basis, via the existing
+   Mixer API (see: [ReportRequest](https://github.com/istio/api/blob/master/mixer/v1/report.proto#L23)).
+1. Publishing the set of schemas for the telemetry data generated by its 
+   component in a *discoverable* fashion.
+
+In order to prevent a never-ending loop of Reports, Mixer will be configured to
+drop ReportRequests received where the source of the traffic being reported is
+Mixer itself.
```

Comment:
this would happen in cases when there was a proxy in front of a logging/metrics backend, and that proxy was also reporting to mixer.